Error deleting channels

**Describe the bug**
When trying to delete a channel, it throws an error, the request is successful but there seems to be a response parsing error which makes it impossible for the delete channel request callback to be successful. I get the following error;

```E/LoggingTracker: logging caught exception
    io.getstream.chat.android.client.errors.ChatParsingError: custom object deserialisation error of class io.getstream.chat.android.client.models.Channel
        at io.getstream.chat.android.client.parser.adapters.CustomObjectGsonAdapter.read(CustomObjectGsonAdapter.kt:100)
        at io.getstream.chat.android.client.parser.adapters.CustomObjectGsonAdapter.read(CustomObjectGsonAdapter.kt:16)
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:131)
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:222)
        at retrofit2.converter.gson.GsonResponseBodyConverter.convert(GsonResponseBodyConverter.java:40)
        at retrofit2.converter.gson.GsonResponseBodyConverter.convert(GsonResponseBodyConverter.java:27)
        at retrofit2.OkHttpCall.parseResponse(OkHttpCall.java:243)
        at retrofit2.OkHttpCall$1.onResponse(OkHttpCall.java:153)
        at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:923)
     Caused by: io.getstream.chat.android.client.errors.ChatParsingError: custom object deserialisation error of class io.getstream.chat.android.client.models.User
        at io.getstream.chat.android.client.parser.adapters.CustomObjectGsonAdapter.read(CustomObjectGsonAdapter.kt:100)
        at io.getstream.chat.android.client.parser.adapters.CustomObjectGsonAdapter.read(CustomObjectGsonAdapter.kt:16)
        at com.google.gson.TypeAdapter.fromJson(TypeAdapter.java:260)
        at com.google.gson.TypeAdapter.fromJson(TypeAdapter.java:273)
        at io.getstream.chat.android.client.parser.adapters.CustomObjectGsonAdapter.read(CustomObjectGsonAdapter.kt:86)
        at io.getstream.chat.android.client.parser.adapters.CustomObjectGsonAdapter.read(CustomObjectGsonAdapter.kt:16) 
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:131) 
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:222) 
        at retrofit2.converter.gson.GsonResponseBodyConverter.convert(GsonResponseBodyConverter.java:40) 
        at retrofit2.converter.gson.GsonResponseBodyConverter.convert(GsonResponseBodyConverter.java:27) 
        at retrofit2.OkHttpCall.parseResponse(OkHttpCall.java:243) 
        at retrofit2.OkHttpCall$1.onResponse(OkHttpCall.java:153) 
        at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:923) 
     Caused by: java.lang.NullPointerException: delegateMapAdapter.read(reader) must not be null
        at io.getstream.chat.android.client.parser.MapAdapter.read(MapAdapter.kt:8)
        at io.getstream.chat.android.client.parser.MapAdapter.read(MapAdapter.kt:7)
```
This makes the enqueue callback is not successful and


**SDK version**
- 4.4.9

**To Reproduce**
Try to delete a channel and enqueue. I'm using the following method to try to delete
```kotlin
ChatClient.instance().channel(channelCid).delete().enqueue {
            ...
 }
```

**Expected behavior**
When request to delete a channel is successful, the enqueue callback task should be successful and return the right response, instead it returns the cause of the error as `io.getstream.chat.android.client.errors.ChatParsingError: custom object deserialisation error of class io.getstream.chat.android.client.models.Channel` given the error log above

--- a/UNRELEASED_CHANGELOG.md
+++ b/UNRELEASED_CHANGELOG.md
@@ -7,6 +7,7 @@
 ## stream-chat-android-client
 - Add filtering non image attachments in ChatClient::getImageAttachments
 - Add a `channel` property to `notification.message_new` events
+- Fix deleting channel error
 
 ## stream-chat-android-offline
 - Add LeaveChannel use case

--- a/stream-chat-android-client/src/main/java/io/getstream/chat/android/client/parser/MapAdapter.kt
+++ b/stream-chat-android-client/src/main/java/io/getstream/chat/android/client/parser/MapAdapter.kt
@@ -5,7 +5,7 @@ import com.google.gson.stream.JsonReader
 import com.google.gson.stream.JsonWriter
 
 internal class MapAdapter(private val delegateMapAdapter: TypeAdapter<Map<*, *>>) : TypeAdapter<Map<*, *>>() {
-    override fun read(reader: JsonReader?): Map<*, *> = delegateMapAdapter.read(reader)
+    override fun read(reader: JsonReader?): Map<*, *>? = delegateMapAdapter.read(reader)
     override fun write(writer: JsonWriter?, value: Map<*, *>?) =
         delegateMapAdapter.write(writer, value?.filterValues { it != null }?.takeIf { it.isNotEmpty() })
 }

